questionnaire/routes.js

```diff
@@ -66,7 +66,11 @@ router
             // There can only every be one "answers" block per section
             // TODO: handle multiple attempts to "create" answers
             const answers = req.body.data.attributes;
-            const questionnaireService = createQuestionnaireService({logger: req.log});
+            const questionnaireService = createQuestionnaireService({
+                logger: req.log,
+                apiVersion: req.get('Dcs-Api-Version'),
+                ownerId: req.get('On-Behalf-Of')
```

Comment:
I think there's two solutions. Either:
```
            if (apiVersion === '2023-05-17' && sectionId !== 'system') {
                await db.updateQuestionnaireByOwner(questionnaireId, answeredQuestionnaire);
            } else {
                await db.updateQuestionnaire(questionnaireId, answeredQuestionnaire);
            }
```            
1. The above works by letting system answers continue to be updated without an owner. The 'system' answer endpoint is protected because it's got a JWT with a specific scope, so a bad actor would need a new JWT to manipulate this

or

```
            if (apiVersion !== '2023-05-17' || ownerId === 'some-id-we-give-IS') {
                await db.updateQuestionnaire(questionnaireId, answeredQuestionnaire);
            } else {
                await db.updateQuestionnaireByOwner(questionnaireId, answeredQuestionnaire);
            }
```            
2. Same idea at the DB level, only we give IS an ID for the request and the DCS knows about it and what to do with it.

My preference is 1, only concern is if this is an abuse of a JWT or no? Worth pointing out as well, at some point v1 will be deprecated and we can simplify this IF logic. At that point I'd probably write a specific 'update system answer' query, rather than leaving it as a 'update everything'  query. Happy to hear thoughts/alternatives?

Reply:
As discussed, v1 is to be used instead